Sources/Core/Extensions/Other/XCTestCase+URLRequest.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  XCTestCase+URLRequest.swift
+//  PovioKit
+//
+//  Created by Gentian Barileva on 09/09/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import XCTest
+
+public extension XCTestCase {
+  func XCTAssertEqualURLRequests(_ lhs: URLRequest?, _ rhs: URLRequest?, file: StaticString = #filePath, line: UInt = #line) {
```

Comment:
Maybe we can override the existing ‘XCAssertEqual’ instead? I would also rather see non-optional arguments.

Reply:
> Maybe we can override the existing ‘XCAssertEqual’ instead? I would also rather see non-optional arguments.

Hey Toni, Thanks for the feedback.

I tried overloading the method, but I'm getting some errors. Could you give me an example of how it would look like? It would greatly help me out.

Regarding the optional arguments, I have two points that I would like to mention:

The first one would be that sending optional values is already supported by XCTest methods, so it wouldn't fall outside their practices, and the second one is more related to how this method came to life.
The particular use case was for testing a URLSession client. Since it accepts URLRequest, I had to intercept the request and assert that it was the correct one. By that flow, if the implementation is wrong, the URLRequest can be optional. You can argue that you can assert unwrap the optional value, but I deemed it unnecessary since you can already XCTAssert optional and non-optional values.

Reply:
Let's keep optional arguments, I agree.

> Could you give me an example of how it would look like

This compiles for me:

```swift
func XCTAssertEqual(_ lhs: URLRequest?, _ rhs: URLRequest?, file: StaticString = #filePath, line: UInt = #line) {
  XCTAssertTrue(lhs == rhs)
  guard let lhs = lhs, let rhs = rhs else { return }
  XCTAssertEqual(lhs.url, rhs.url, "url", file: file, line: line)
  XCTAssertEqual(lhs.httpMethod, rhs.httpMethod, "httpMethod", file: file, line: line)
  XCTAssertEqual(lhs.httpBody, rhs.httpBody, "httpBody", file: file, line: line)
  XCTAssertEqual(lhs.allHTTPHeaderFields, rhs.allHTTPHeaderFields, "allHTTPHeaderFields", file: file, line: line)
}
```

Reply:
If I try to add it as an extension to XCTestCase then it gives errors such as: 
`Use of 'XCTAssertEqual' refers to instance method rather than global function 'XCTAssertEqual(_:_:_:file:line:)' in module 'XCTest'`

Reply:
Just make it a global function (other `XCTest...` methods are global as well).